string[] stages
float64[] times
